src/Raven.Server/Rachis/Follower.cs

```diff
@@ -32,8 +33,9 @@ public Follower(RachisConsensus engine, long term, RemoteConnection remoteConnec
             _engine = engine;
             _connection = remoteConnection;
             _term = term;
-            
-            _debugName = $"Follower in term {_term}";
+
+            // this will give us a unique identifier for this follower
+            _debugName = $"Follower in term {_term} started at {SystemTime.UtcNow:s}";
```

Comment:
This isn't actually going to generate unique identifier. This is already racing, so this will happen if you have this in the same second. That is pretty much _guaranteed_ to be the case. You need to use an actual unique value, best to use something like `Interlocked.Increment` on a static value here.